react-multiple-pages/my-app/src/App.tsx

```diff
@@ -0,0 +1,20 @@
+import { About } from './pages/About';
+import { Dashboard } from './pages/Dashboard';
+import { Header } from './components/Header';
+import { NotFound } from './pages/NotFound';
+import { Details } from './pages/Details';
+import { Route, Routes } from 'react-router-dom';
+
+export function App() {
+  return (
+    <>
+      <Routes>
+        <Route path="/" element={<Header />} />
```

Comment:
This is supposed to wrap around the other routes. Then you use Outlet in the Header to indicate where you want those route components to display.
`<Route path="/" element={<Header />}> ...other routes... </Route>`